master/buildbot/process/factory.py

```diff
@@ -92,19 +92,13 @@ def __init__(self, source, configure="./configure",
                  configureEnv=None,
                  configureFlags=None,
                  reconf=None,
-                 compile=None,
-                 test=None,
-                 distcheck=None):
+                 compile=["make", "all"],
```

Comment:
I'm not changing the semantics of compile, distcheck and test steps: currently they already receive a list of values. The fix consists in initialize the variables or skip the step if None is set, currently setting to None will keep the step (and use default command).

The manual says None should be set to skip the step. Other factories (e.g. Distutils and Trial) uses same semantic, i.e., set None to disable a step. Also IMHO setting False is more suitable if the variable expects a boolean value.